### PR TITLE
Slaunch conditional debug

### PR DESCRIPTION
**Pull request by miczyg1**

### CODE REVIEW

**Comment by miczyg1** [comment]
If grub.cfg contains `set debug=slaunch` then the debug information is printed. Otherwise the output is silent. Additionally grub_dprintf automatically adds the file and line number the print originates from. Because of that I have removed the function names and lien numbers from prints.

File: grub-core/loader/i386/slaunch.c [hunk 1]
```diff
@@ -91,20 +91,20 @@ grub_cmd_slaunch (grub_command_t cmd __attribute__ ((unused)),
     return grub_error (GRUB_ERR_BAD_ARGUMENT, N_("argument expected"));
 
   /* Should be executing on the BSP  */
-  grub_printf("%s:%d: about to do rdmsr: ", __FUNCTION__, __LINE__);
+  grub_dprintf ("slaunch", "about to do rdmsr: ");
```
Comment (krystian-hebel) [inline grub-core/loader/i386/slaunch.c]: @miczyg1 It seems that output is buffered and flushed on new line, so if `grub_rdmsr()` fails, this line will not be printed. Can you add `\r\n` here, and perhaps indent the following "OK"? Or just drop those lines altogether, these really shouldn't fail now and do not provide any useful information anymore.

File: grub-core/loader/i386/slaunch.c [hunk 2]
```diff
@@ -91,20 +91,20 @@ grub_cmd_slaunch (grub_command_t cmd __attribute__ ((unused)),
...
   msr_value = grub_rdmsr (GRUB_MSR_X86_APICBASE);
   if (! (msr_value & GRUB_MSR_X86_APICBASE_BSP))
     return grub_error (GRUB_ERR_BAD_DEVICE, N_("secure launch must run on BSP"));
 
-  grub_printf("OK\r\n");
-  grub_printf("%s:%d: about to do cpuid ", __FUNCTION__, __LINE__);
+  grub_dprintf ("slaunch", "OK\r\n");
+  grub_dprintf ("slaunch", "about to do cpuid ");
```
Comment (krystian-hebel) [inline grub-core/loader/i386/slaunch.c]: As above

File: grub-core/loader/i386/slaunch.c [hunk 3]
```diff
@@ -91,20 +91,20 @@ grub_cmd_slaunch (grub_command_t cmd __attribute__ ((unused)),
...
   if (! grub_cpu_is_cpuid_supported ())
     return grub_error (GRUB_ERR_UNKNOWN_DEVICE, N_("CPUID not supported"));
 
-  grub_printf("(supported) ");
+  grub_dprintf ("slaunch", "(supported) ");
```
Comment (krystian-hebel) [inline grub-core/loader/i386/slaunch.c]: As above

File: grub-core/loader/i386/slaunch.c [hunk 4]
```diff
@@ -91,20 +91,20 @@ grub_cmd_slaunch (grub_command_t cmd __attribute__ ((unused)),
...
   grub_cpuid (0, eax, manufacturer[0], manufacturer[2], manufacturer[1]);
 
-  grub_printf("OK\r\n");
+  grub_dprintf ("slaunch", "OK\r\n");
```
Comment (krystian-hebel) [inline grub-core/loader/i386/slaunch.c]: As above

**Review by krystian-hebel — CHANGES_REQUESTED** [review]

File: grub-core/loader/i386/slaunch.c (hunk 1 above)
Comment (miczyg1) [reply grub-core/loader/i386/slaunch.c]: Removed these

File: grub-core/loader/i386/slaunch.c (hunk 3 above)
Comment (miczyg1) [reply grub-core/loader/i386/slaunch.c]: As above, removed.

File: grub-core/loader/i386/slaunch.c (hunk 4 above)
Comment (miczyg1) [reply grub-core/loader/i386/slaunch.c]: As above, removed.

File: grub-core/loader/i386/slaunch.c (hunk 2 above)
Comment (miczyg1) [reply grub-core/loader/i386/slaunch.c]: As above, removed.

**Review by krystian-hebel — APPROVED** [review]

**Pull request merged**